### ISSUE TITLE
Error after edit Excel file

### ISSUE THREAD

**Issue by anhtester** [sskorol/test-data-supplier#112]
Excel file old: 
[ClientsDataExcelOLD.xlsx](https://github.com/sskorol/test-data-supplier/files/8409459/ClientsDataExcelOLD.xlsx)

Excel file after edit: 
[ClientsDataExcel.xlsx](https://github.com/sskorol/test-data-supplier/files/8409460/ClientsDataExcel.xlsx)

I can not read excel file after edit.

This my class define: (.txt convert to .java)
[SignIn.txt](https://github.com/sskorol/test-data-supplier/files/8409498/SignIn.txt)
[Client.txt](https://github.com/sskorol/test-data-supplier/files/8409464/Client.txt)

It doesn't give a clear error message so it's hard for me to handle it.

![image](https://user-images.githubusercontent.com/87883620/161540349-66e5c5d5-51e8-4080-afcb-349111f513d3.png)

**Comment by sskorol** [comment]
See https://github.com/sskorol/test-data-supplier/issues/111#issuecomment-1087810478. Seems like you are trying to compile your project with java 8 that's not supported by the recent test-data-supplier.

P.S. There's no error as the test is just skipped. So you should check your report for details.

**Comment by anhtester** [comment]
There is a problem with the filter
please check above excel file of me

![image](https://user-images.githubusercontent.com/87883620/161609522-0cd0584f-b6f5-4d6a-bab1-90b4a15ccf0e.png)

**Comment by sskorol** [comment]
Paste a full stracktrace as a formatted text here instead of a screenshots please.

**Comment by sskorol** [comment]
Also note that test-data-supplier just wraps https://github.com/creditdatamw/zerocell to provide a convenient way of getting Excel data following data-driven approach. If you your excel file can't be parsed for some reason, it's likely reproducible in ZeroCell as well. So I'd anaylize a stacktrace / test results report first (as skipped tests are still generating stacktraces; testng just doesn't print such them to CLI output), and check the limitations of ZeroCell. Maybe you have a syntax error or the features you use in Excel are not supported by ZeroCell. Anyway, it's hard to say by a part of a screenshot what's going on.

**Comment by anhtester** [comment]
I edited excel and name column then it work. But Filter error. 

Error message not unclear

**Comment by sskorol** [comment]
I've already mentioned above that testng doesn't print stacktraces for skipped tests in IDE logs. Check your test results report for the exception.

**Comment by anhtester** [comment]
I found the filter unstable. You can re check it.
Sample with my file above.

**Comment by sskorol** [comment]
@anhtester the reason why it doesn't work for you is NPE thrown from data provider. The reason why you can't see it in logs is TestNG's logic for the exceptions thrown from configuration methods (it just suppresses them and skips the test). The reason for NPE is an attempt to call `equals` method on null field (accessed via `getTestCaseName()`). The reason why this field is null is a set of empty rows in your excel spreadsheets. The reason why empty rows are not skipped is probably a bug in ZeroCell, as test-data-supplier uses `skipEmptyRows(true)` internally. So you may want to raise it in their repo if it's important for you.

How to fix this particular issue? Either check if objects are nullable or do the opposite comparison: `method.getName().equals(...)`. 

To summarize:
- it's not a test-data-supplier issue;
- that's how TestNG works in terms of exceptions handling in data providers;
- that's how Java works in terms of handling uninitialized objects;
- that's how ZeroCell works with empty rows.

Next time, please provide a minimal repo to reproduce the issue + text logs (screenshots are not informative), or just spend another 10 mins debugging your code before creating an issue. NPEs' handling is your own responsibility.

**Comment by anhtester** [comment]
Okay. Thank you so much !!
